### PR TITLE
doc: timers

### PR DESCRIPTION
**Pull request by ggreif**
Adds a few examples how the timer mechanisms can be used.

### CODE REVIEW

File: doc/md/timers.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Timers
+
+Internet Computer canisters can set an arbitrary number of single-expiration or recurring timers. See the `Timers.mo` module in the base library.
+
+The underlying mechanism is a [canister global timer](https://internetcomputer.org/docs/current/references/ic-interface-spec#timer) that is reprogrammed according to an internally kept priority queue.
+
+The timer mechanism can be disabled completely by passing the `-no-timer` flag to `moc`.
+
+When utmost control about the global timer is necessary, an actor can receive global timer expiry messages by declaring a `system` function, named `timer`, with one argument (to set the global timer), returning a future of unit type (`async ()`). In this case the `Timers.mo` base library module is not functional and shouldn't be used.
```
Comment (crusso) [inline doc/md/timers.md:9]:
```suggestion
When lower-level access to the canister global timer is desired, an actor can elect to receive timer expiry messages by declaring a `system` function, named `timer`. The function takes one argument (to re-set the global timer), and returns a future of unit type (`async ()`).
If the `timer` system method is declared, the `Timers.mo` base library module may not function correctly and should not be used.
```

File: doc/md/timers.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Timers
+
+Internet Computer canisters can set an arbitrary number of single-expiration or recurring timers. See the `Timers.mo` module in the base library.
+
+The underlying mechanism is a [canister global timer](https://internetcomputer.org/docs/current/references/ic-interface-spec#timer) that is reprogrammed according to an internally kept priority queue.
+
+The timer mechanism can be disabled completely by passing the `-no-timer` flag to `moc`.
```
Comment (crusso) [inline doc/md/timers.md:7]: Is the flag documented in compiler.md (or whatever it's called)?

**Review by crusso — CHANGES_REQUESTED** [review]

File: doc/md/timers.md (hunk 2 above)
Comment (ggreif) [reply doc/md/timers.md:7]: Adding...

File: doc/md/timers.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+When lower-level access to the canister global timer is desired, an actor can elect to receive timer expiry messages by declaring a `system` function, named `timer`. The function takes one argument (to re-set the global timer), and returns a future of unit type (`async ()`).
+If the `timer` system method is declared, the `Timers.mo` base library module may not function correctly and should not be used.
+
+The following example of a global timer expiration callback gets called immediately after
+the canister starts (i.e. after install) and periodically every twenty seconds thereafter:
+
+``` motoko
```
Comment (crusso) [inline doc/md/timers.md:20]:
```suggestion
``` motoko no-repl
```

File: doc/md/timers.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Timers
+
+Internet Computer canisters can set an arbitrary number of single-expiration or recurring timers. See the `Timers.mo` module in the base library.
+
+A simple, contrived example is a periodic reminder, that logs a new-year's message:
+
+``` motoko file=./examples/Reminder.mo
```
Comment (crusso) [inline doc/md/timers.md:7]:
```suggestion
``` motoko no-repl file=./examples/Reminder.mo
```

Comment (crusso) [reply doc/md/timers.md:7]: Could you add a no-repl to the hearbeat example too, while you've got it checked out?

**Review by crusso — APPROVED** [review]
LGTM. Tried in playground but Timer.mo not available yet. I guess we can check with the next release.

**Comment by crusso** [comment]
Thanks for doing it! Cute example, though they'll have to wait a year to see it have no effect in the playground..

I guess the other option would be to have a regularly incremented counter whose values can be observed.

**Pull request merged**